website_and_docs/content/documentation/grid/advanced_features/customize_node.en.md

```diff
@@ -0,0 +1,206 @@
+---
+title: "Customizing a Node"
+linkTitle: "Customize Node"
+weight: 4
+aliases: [
+"/documentation/en/grid/grid_4/customize_node/",
+"/documentation/grid/advanced_features/customize_node/"
+]
+---
+
+
+### How to customize a Node
+
+There are times when we would like a node to be customized to our needs. 
+
+For e.g., we may like to do some additional setup before a test case begins execution and some clean-up after a test case runs to completion.
+
+Following steps can be followed for this:
+
+* Create a class that extends `org.openqa.selenium.grid.node.Node`
+* Add a static method (this will be our factory method) to the newly created class whose signature looks like this: 
+
+  `public static Node create(Config config)`. Here:
+
+    * `Node` is of type `org.openqa.selenium.grid.node.Node`
+    * `Config` is of type `org.openqa.selenium.grid.config.Config`
+* Within this factory method, include logic for creating your new Class.
+* To wire in this new customized logic into the hub, start the node and pass in the fully qualified class name of the above class to the argument `--node-implementation`
+
+Let's see an example of all this:
+
+1. Create a sample project using your favourite build tool (**Maven**|**Gradle**).
+2. Add the below two dependencies to your sample project.
+  * [org.seleniumhq.selenium/selenium-java](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java)
+  * [org.seleniumhq.selenium/selenium-grid](https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-grid/4.6.0)
```

Comment:
I believe only the Grid dependency is needed, isn't it?

Reply:
Also, no need to have the URL with the exact version.

Reply:
@diemol - 

> I believe only the Grid dependency is needed, isn't it?

Yes and No to be honest. If one is building an uber/fat jar that one intends to use as the selenium server binary, then merely depending on `selenium-grid` will give only chrome capabilities. Only by depending on `selenium-java` will they be able to get support for the other browser flavors. I intend to send more PRs on this page which tackles both the use cases.

> Also, no need to have the URL with the exact version.

That was a typo :)